File: trunk/simulations-java/simulations/fractions/src/edu/colorado/phet/fractions/common/util/immutable/Dimension2D.java
// Copyright 2002-2012, University of Colorado
package edu.colorado.phet.fractions.common.util.immutable;

import lombok.Data;

import edu.colorado.phet.common.phetcommon.view.Dimension2DDouble;

/**
 * Immutable class for a dimension (width and height).  Uses Lombok to generate equals, hashcode.
 *
 * @author Sam Reid
 */
public @Data class Dimension2D {
    public final double width;
    public final double height;

    public Dimension2DDouble toDimension2DDouble() { return new Dimension2DDouble( width, height ); }
}

File: team/cmalley/html5/canvas-resize/js/math/Dimension2D.js
// Copyright 2002-2012, University of Colorado

/**
 * 2D dimension.
 *
 * @author Chris Malley (PixelZoom, Inc.)
 */
define( [],
        function () {

            /**
             * @constructor
             * @param {Number} width
             * @param {Number} height
             */
            function Dimension2D( width, height ) {
                this.width = width;
                this.height = height;
            }

             // @return {String}
            Dimension2D.prototype.toString = function () {
                return "[Dimension2D (width=" + this.width + " height=" + this.height + ")]";
            };

            return Dimension2D;
        } );
